model/src/exec_env.rs

```diff
@@ -180,4 +181,26 @@ impl<T: Env> EnvHandler<T> {
     pub fn block_number(&self) -> T::BlockNumber {
         T::block_number()
     }
+
+    /// Calls a remote smart contract
```

Comment:
Also please extend the documentation here since this is a very fragile part of the env API with many open questions. Try asking yourself what you would ask yourself about this and try to cover those questions in the docs. Same for the other API.